linux/net/rina/ipcps/shim-eth-vlan.c

```diff
@@ -270,11 +275,6 @@ find_flow_by_gpa(struct ipcp_instance_data * data,
 
 static bool vlan_id_is_ok(uint16_t vlan_id)
 {
-        if (vlan_id > 4095 /* 0xFFF */) {
-                /* Out of bounds */
-                return false;
-        }
-
         ASSERT(vlan_id <= 4095);
 
```

Comment:
This should remain an if() check, since it is just parsing userspace, and should not crash the kernel, but exit gracefully if the VLAN id is out of bounds.